File: clients/java/src/main/java/com/svix/coyote/apis/RateLimiter.java
// this file is @generated
package com.svix.coyote.apis;

import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.svix.coyote.ApiException;
import com.svix.coyote.HttpClient;
import com.svix.coyote.Utils;
import java.io.IOException;
import java.util.Collections;
import java.util.HashMap;
import java.util.Map;
import java.util.Set;
import okhttp3.Headers;
import okhttp3.HttpUrl;
import com.svix.coyote.models.RateLimiterCheckIn;
import com.svix.coyote.models.RateLimiterCheckOut;
import com.svix.coyote.models.RateLimiterGetRemainingIn;
import com.svix.coyote.models.RateLimiterGetRemainingOut;

public class RateLimiter {
    private final HttpClient client;

    public RateLimiter(HttpClient client) {
        this.client = client;
    }

    /** Rate Limiter Check and Consume */
    public RateLimiterCheckOut limit(
        final RateLimiterCheckIn rateLimiterCheckIn
    ) throws IOException, ApiException {
        HttpUrl.Builder url = this.client.newUrlBuilder().encodedPath("/api/v1/rate-limiter/limit");
        return this.client.executeRequest(
            "POST",
            url.build(),
            null,
            rateLimiterCheckIn,
            RateLimiterCheckOut.class
            );
    }

    /** Rate Limiter Get Remaining */
    public RateLimiterGetRemainingOut getRemaining(
        final RateLimiterGetRemainingIn rateLimiterGetRemainingIn
    ) throws IOException, ApiException {
        HttpUrl.Builder url = this.client.newUrlBuilder().encodedPath("/api/v1/rate-limiter/get-remaining");
        return this.client.executeRequest(
            "POST",
            url.build(),
            null,
            rateLimiterGetRemainingIn,
            RateLimiterGetRemainingOut.class
            );
    }
}

File: clients/python/coyote/apis/rate_limiter.py
# This file is @generated

from ..internal.api_common import ApiBase
from ..models import (
    RateLimiterCheckIn,
    RateLimiterCheckOut,
    RateLimiterGetRemainingIn,
    RateLimiterGetRemainingOut,
)


class RateLimiterAsync(ApiBase):
    async def limit(
        self,
        rate_limiter_check_in: RateLimiterCheckIn,
    ) -> RateLimiterCheckOut:
        """Rate Limiter Check and Consume"""
        response = await self._request_asyncio(
            method="post",
            path="/api/v1/rate-limiter/limit",
            path_params={},
            json_body=rate_limiter_check_in.model_dump_json(
                exclude_unset=True, by_alias=True
            ),
        )
        return RateLimiterCheckOut.model_validate(response.json())

    async def get_remaining(
        self,
        rate_limiter_get_remaining_in: RateLimiterGetRemainingIn,
    ) -> RateLimiterGetRemainingOut:
        """Rate Limiter Get Remaining"""
        response = await self._request_asyncio(
            method="post",
            path="/api/v1/rate-limiter/get-remaining",
            path_params={},
            json_body=rate_limiter_get_remaining_in.model_dump_json(
                exclude_unset=True, by_alias=True
            ),
        )
        return RateLimiterGetRemainingOut.model_validate(response.json())


class RateLimiter(ApiBase):
    def limit(
        self,
        rate_limiter_check_in: RateLimiterCheckIn,
    ) -> RateLimiterCheckOut:
        """Rate Limiter Check and Consume"""
        response = self._request_sync(
            method="post",
            path="/api/v1/rate-limiter/limit",
            path_params={},
            json_body=rate_limiter_check_in.model_dump_json(
                exclude_unset=True, by_alias=True
            ),
        )
        return RateLimiterCheckOut.model_validate(response.json())

    def get_remaining(
        self,
        rate_limiter_get_remaining_in: RateLimiterGetRemainingIn,
    ) -> RateLimiterGetRemainingOut:
        """Rate Limiter Get Remaining"""
        response = self._request_sync(
            method="post",
            path="/api/v1/rate-limiter/get-remaining",
            path_params={},
            json_body=rate_limiter_get_remaining_in.model_dump_json(
                exclude_unset=True, by_alias=True
            ),
        )
        return RateLimiterGetRemainingOut.model_validate(response.json())
